src/test/java/org/cactoos/list/JoinedTest.java

```diff
@@ -435,4 +439,25 @@ public void itemAndList() {
             )
         ).affirm();
     }
+
+    @Test
+    public void infersCorrectly() {
```

Comment:
@andreoss please, add more tests, for a `map` and the decorators where were changed.